Lib/test/test_subprocess.py

```diff
@@ -1129,6 +1129,28 @@ def test_universal_newlines_communicate_encodings(self):
             stdout, stderr = popen.communicate(input='')
             self.assertEqual(stdout, '1\n2\n3\n4')
 
+    @unittest.skipIf(mswindows, "behavior currently not supported on Windows")
+    def test_universal_newlines_timeout(self):
+        with self.assertRaises(subprocess.TimeoutExpired) as c:
+            p = subprocess.run(
+                [
+                    sys.executable, "-c",
+                    "import sys, time;"
+                    r"sys.stderr.buffer.write(b'foo \xc2\xa4 bar');"
+                    "sys.stderr.buffer.flush();"
+                    r"sys.stdout.buffer.write(b'foo \xc2');"
+                    "sys.stdout.buffer.flush();"
+                    "time.sleep(0.1);"
+                    r"sys.stdout.buffer.write(b'\xa4 bar');"
+                    "sys.stdout.buffer.flush();"
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                universal_newlines=True,
+                timeout=0.05)
```

Comment:
Slower buildbots may need more time. See `ProcessTestCase.test_check_output_timeout()`. It uses a timeout of 3 seconds. 

The final `sys.stdout.buffer.write(b'\xa4 bar')` isn't needed. The timeout condition with an incomplete write is all that's tested.

Reply:
Thanks, changes made (maybe this will address the mac failure?).